tools/third_party.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# This script contains helper functions to work with the third_party subrepo.
+
+import os
+import sys
+from os import path
+from util import find_exts, make_env, remove_and_symlink, rmtree, root_path, run
+
+
+# Helper function that returns the full path to a subpath of the repo root.
+def root(*subpath_parts):
+    return path.normpath(path.join(root_path, *subpath_parts))
+
+
+# Helper function that returns the full path to a file/dir in third_party.
+def tp(*subpath_parts):
+    return root("third_party", *subpath_parts)
+
+
+third_party_path = tp()
+depot_tools_path = tp("depot_tools")
+rust_crates_path = tp("rust_crates")
+
+
+# This function creates or modifies an environment so that it matches the
+# expectations of various google tools (gn, gclient, etc).
+def google_env(env=None, merge_env={}, depot_tools_path=depot_tools_path):
+    env = make_env(env=env, merge_env=merge_env)
+    # Depot_tools to be in the PATH, before Python.
+    path_prefix = depot_tools_path + os.path.pathsep
+    if not env['PATH'].startswith(path_prefix):
+        env['PATH'] = path_prefix + env['PATH']
+    # We're not using Google's internal infrastructure.
+    if os.name == 'nt' and not 'DEPOT_TOOLS_WIN_TOOLCHAIN' in env:
+        env['DEPOT_TOOLS_WIN_TOOLCHAIN'] = "0"
+    return env
+
+
+def fix_symlinks():
+    # Ensure the third_party directory exists.
+    try:
+        os.makedirs(third_party_path)
+    except:
+        pass
+
+    # Make symlinks to Yarn metadata living in the root repo.
+    remove_and_symlink("../package.json", tp("package.json"))
+    remove_and_symlink("../yarn.lock", tp("yarn.lock"))
+
+    # TODO(ry) Is it possible to remove these symlinks?
+    remove_and_symlink("v8/third_party/googletest", tp("googletest"), True)
+    remove_and_symlink("v8/third_party/jinja2", tp("jinja2"), True)
+    remove_and_symlink("v8/third_party/llvm-build", tp("llvm-build"), True)
+    remove_and_symlink("v8/third_party/markupsafe", tp("markupsafe"), True)
+
+    # On Windows, git doesn't create the right type of symlink if the symlink
+    # and it's target are in different repos. Here we fix the symlinks that exist
+    # in the root repo while their target is in the third_party repo.
+    remove_and_symlink("third_party/node_modules", root("node_modules"), True)
+    remove_and_symlink("third_party/v8/build", root("build"), True)
+    remove_and_symlink("third_party/v8/buildtools", root("buildtools"), True)
+    remove_and_symlink("third_party/v8/build_overrides",
+                       root("build_overrides"), True)
+    remove_and_symlink("third_party/v8/testing", root("testing"), True)
```

Comment:
These lines have the first argument prefixed with "third_party/", but the cwd should already be in third_party. I think there's an error here.

Reply:
The first argument is the symlink target. It's not affected by chdir, symlinks are resolved from the directory that contains the symlink